crates/flox/src/commands/channel.rs

```diff
@@ -132,62 +156,229 @@ fn construct_search_params(search_term: &str, flox: &Flox) -> Result<SearchParam
+            // Remove this input so this (input, package) pair is never seen again
+            if inputs.remove(&d.input) {
+                deduped_display_items.push(d.clone());
+            }
+            if inputs.is_empty() {
+                package_to_inputs.remove(&d.package);
+            }
+        }
+    }
+
+    deduped_display_items
+}
+
+/// Show detailed package information
+#[derive(Bpaf, Clone)]
+pub struct Show {
+    /// The package to show detailed information about. Must be an exact match
+    /// for a package name e.g. something copy-pasted from the output of `flox search`.
+    #[bpaf(positional("search-term"))]
+    pub search_term: String,
+}
+
+impl Show {
+    pub async fn handle(self, flox: Flox) -> Result<()> {
+        let search_params = construct_show_params(&self.search_term, &flox)?;
+
+        let (search_results, exit_status) = do_search(&search_params)?;
+
+        if search_results.results.is_empty() {
+            bail!("no packages matched this search term: {}", self.search_term);
+        }
+        // Render what we have no matter what, then indicate whether we encountered an error.
+        // FIXME: We may have warnings on `stderr` even with a successful call to `pkgdb`.
+        //        We aren't checking that at all at the moment because better overall error handling
+        //        is coming in a later PR.
+        render_show(search_results.results.as_slice())?;
+        if exit_status.success() {
+            Ok(())
+        } else {
+            bail!(
+                "pkgdb exited with status code: {}",
+                exit_status.code().unwrap_or(-1),
+            );
+        }
+    }
+}
+
+fn construct_show_params(search_term: &str, flox: &Flox) -> Result<SearchParams> {
+    let parts = search_term
+        .split(SEPARATOR)
+        .map(String::from)
+        .collect::<Vec<_>>();
+    let (input_name, package_name) = match parts.len() {
+        1 => Ok((None, Some(parts[0].clone()))),
+        2 => Ok((Some(parts[0].clone()), Some(parts[1].clone()))),
+        _ => Err(ShowError::InvalidSearchTerm(search_term.into())),
+    }?;
+
+    // If we're given a specific input to search, only search that one,
+    // otherwise build the whole list of inputs to search
+    let (inputs, priority) = if let Some(input_name) = input_name {
+        let Some(reg_input) = flox
+            .channels
+            .iter()
+            .find(|entry| entry.from.id == input_name)
+            .map(|entry| RegistryInput {
+                from: entry.to.clone(),
+                subtrees: None,
+                stabilities: None,
+            })
+        else {
+            bail!("registry did not contain an input named '{}'", input_name)
+        };
+        let mut inputs = HashMap::new();
+        inputs.insert(input_name.clone(), reg_input);
+        (inputs, vec![input_name])
+    } else {
+        collect_inputs(flox)
```

Comment:
- `collect_inputs` requires some guess work reg. what it's doing, there might be a better name. We should at least document it.
- this `if let` construct could be folded into collect inputs, its doing pretty much the same thing, just filtering more finely. It's kinda odd to have it factored out in only branch

Reply:
`collect_inputs` is a hack to be used until we actually have a manifest, so it will disappear in the near future. For that reason I wouldn't worry about this particular problem too much.